src/ak_schema/schema/ak_specimens.yaml

```diff
@@ -19,23 +19,53 @@ classes:
     slots:
       - specimen
 
+  SpecimenProcessing:
+    is_a: PlannedProcess
+    class_uri: OBI:0000094 # material processing
+    slots:
+      - specimen
+
   Assay:
-    is_a: NamedThing
+    is_a: PlannedProcess
     class_uri: OBI:0000070 # assay
     slots:
       - specimen
-      - type
+      - specimen_processing
       - assay_type
       - target_entity_type
-      - value
-      - unit
+
+  ReceptorRepertoireSequencingAssay:
+    is_a: Assay
+    class_uri: OBI:0600047
+    slots:
+      - template_class
+      - template_quality
+      - template_amount
+      - template_amount_unit
+      - library_generation_method
+      - library_generation_protocol
+      - library_generation_kit_version
+      - pcr_target
+      - complete_sequences
+      - physical_linkage
+  BCellReceptorRepertoireSequencingAssay:
+    is_a: ReceptorRepertoireSequencingAssay
+    class_uri: OBI:0002991
+  TCellReceptorRepertoireSequencingAssay:
```

Comment:
This is an example of the ontology class vs LinkML class distinction in my previous comment. `BCellReceptorRepertoireSequencingAssay` and `TCellReceptorRepertoireSequencingAssay` have different ontology ID (`class_uri`s) but the same slots. If that's the case, then my preference would be to use the same `ReceptorRepertoireSequencingAssay` LinkML class but assign different `assay_type` values to capture this difference.

Reply:
Removed `class_uri` from the subclasses, we would then use `type` and `assay_type` to capture this info.

Reply:
These `BCellReceptorRepertoireSequencingAssay` and `TCellReceptorRepertoireSequencingAssay` subclasses have the same slots as their parent class, so I wouldn't bother with them. But if there's a reason to keep them, then I would also like to keep their `class_uri`s, to say which OBI classes they correspond to.

The `assay_type` is still useful, since it can be more specific than the `class_uri`: ontologies have much more detailed class hierarchies than schemas.

Reply:
My understanding was that the `type` field would refer to an AKC LinkML class. If we remove those classes then that would not work, no?

So I would be tempted to keep them (as we do want to differentiate) but add the `class_uri` back?

Reply:
@jamesaoverton I have added back the class_uri field.

Reply:
> These `BCellReceptorRepertoireSequencingAssay` and `TCellReceptorRepertoireSequencingAssay` subclasses have the same slots as their parent class, so I wouldn't bother with them. But if there's a reason to keep them, then I would also like to keep their `class_uri`s, to say which OBI classes they correspond to.

This is where I disagree with the ontology a little. It has these separated as two assays without having an assay that captures both. A 10X single cell can capture both T and B cell together, so I don't like creating two Assay instances for that one actual assay.

Reply:
Yes the OBO ontology isn't perfect in this area. I have modelled it more logically with a single repertoire sequencing class with a separate B-cell and T-cell subclass. I think you are suggesting that there is also a third class which is a combined class. 

One could capture that by having either the combined B-cell/T-cell assay as the parent class (not specific to either B-cell or T-cell) or you could create a third child class that is a combined B-cell/T-cell assay.

Or you could get rid of the subclasses altogether and just capture this using the `assay_type` field. 

Maybe, for starters, we keep it simple and just have `ReceptorRepertoireSequencingAssay` and use the `assay_type` to differentiate.

Reply:
> we keep it simple and just have `ReceptorRepertoireSequencingAssay`

This one. The PCRTarget indicates the locus so we don't need `assay_type` for that. I'm not exactly sure how we want to use `assay_type` for repertoire sequencing.

Reply:
@schristley @jamesaoverton can we resolve this conversation?